Source/Core/Core/AchievementManager.cpp

```diff
@@ -796,15 +804,16 @@ void AchievementManager::HandleAchievementChallengeIndicatorShowEvent(
     const rc_client_event_t* client_event)
 {
   auto& instance = AchievementManager::GetInstance();
-  instance.m_active_challenges[client_event->achievement->badge_name] =
-      &AchievementManager::GetInstance().GetAchievementBadge(client_event->achievement->id, false);
+  instance.m_active_challenges.insert(client_event->achievement->id);
```

Comment:
Should this handle whether the id already existed?

```
const auto [iter, inserted] = instance.m_active_challenges.insert(client_event->achievement->id);
if (inserted)
  instance.m_challenges_updated = true;
```

Reply:
This and the question below are operating under the impression that rc_client knows what it is doing when triggering these events, but I can add a safety check if you'd like.